Calibration/LumiAlCaRecoProducers/plugins/RawPCCProducer.cc

```diff
@@ -40,42 +40,44 @@ class RawPCCProducer : public edm::global::EDProducer<edm::EndLuminosityBlockPro
 };
 
 //--------------------------------------------------------------------------------------------------
 RawPCCProducer::RawPCCProducer(const edm::ParameterSet& iConfig)
     : lumiCorrectionsToken_(esConsumes<edm::Transition::EndLuminosityBlock>()),
-      putToken_{produces<LumiInfo, edm::Transition::EndLuminosityBlock>(
+      modVeto_(
+          iConfig.getParameter<edm::ParameterSet>("RawPCCProducerParameters").getParameter<std::vector<int>>("modVeto")),
+      applyCorr_(iConfig.getParameter<edm::ParameterSet>("RawPCCProducerParameters")
+                     .getUntrackedParameter<bool>("ApplyCorrections", false)),
+      takeAverageValue_(iConfig.getParameter<edm::ParameterSet>("RawPCCProducerParameters")
+                            .getUntrackedParameter<std::string>("OutputValue", std::string("Average"))),
+      putToken_(produces<LumiInfo, edm::Transition::EndLuminosityBlock>(
           iConfig.getParameter<edm::ParameterSet>("RawPCCProducerParameters")
-              .getUntrackedParameter<std::string>("outputProductName", "alcaLumi"))},
-      takeAverageValue_{iConfig.getParameter<edm::ParameterSet>("RawPCCProducerParameters")
-                            .getUntrackedParameter<std::string>("OutputValue", std::string("Totals"))},
-      modVeto_{
-          iConfig.getParameter<edm::ParameterSet>("RawPCCProducerParameters").getParameter<std::vector<int>>("modVeto")},
-      csvOutLabel_{iConfig.getParameter<edm::ParameterSet>("RawPCCProducerParameters")
-                       .getUntrackedParameter<std::string>("label", std::string("rawPCC.csv"))},
-      saveCSVFile_{iConfig.getParameter<edm::ParameterSet>("RawPCCProducerParameters")
-                       .getUntrackedParameter<bool>("saveCSVFile", false)},
-      applyCorr_{iConfig.getParameter<edm::ParameterSet>("RawPCCProducerParameters")
-                     .getUntrackedParameter<bool>("ApplyCorrections", false)} {
+              .getUntrackedParameter<std::string>("outputProductName", "alcaLumi"))),
+      saveCSVFile_(iConfig.getParameter<edm::ParameterSet>("RawPCCProducerParameters")
+                       .getUntrackedParameter<bool>("saveCSVFile", false)),
+      csvOutLabel_(iConfig.getParameter<edm::ParameterSet>("RawPCCProducerParameters")
+                       .getUntrackedParameter<std::string>("label", std::string("rawPCC.csv"))) {
   auto pccSource =
       iConfig.getParameter<edm::ParameterSet>("RawPCCProducerParameters").getParameter<std::string>("inputPccLabel");
   auto prodInst =
       iConfig.getParameter<edm::ParameterSet>("RawPCCProducerParameters").getParameter<std::string>("ProdInst");
-
   pccToken_ = consumes<reco::PixelClusterCounts, edm::InLumi>(edm::InputTag(pccSource, prodInst));
+
+  std::cout << "RawPCCProducer veto list size: " << modVeto_.size() << std::endl;
+  std::cout << "RawPCCProducer applyCorr: " << applyCorr_ << std::endl;
+  std::cout << "RawPCCProducer takeAverage: " << takeAverageValue_.c_str() << std::endl;
```

Comment:
Please move to `edm::LogPrint("RawPCCProducer")` instead of `std::cout`, you can also remove the `std::endl` with that. This is needed for multi-threading

Reply:
even if it's in the constructor so it will be printed only 1 time per stream, do we really need this to be printed in the T0 logs?

Reply:
Ok in that case please use `edm::LogDebug("RawPCCProducer")`

Reply:
`LogDebug` is not part of the `edm` namespace